### PR TITLE
Fixes #2698 - Prevents "Possibly searching too often" error after re-login.

### LINKED ISSUE #2698: Possibly Searching Too Often - keeps changing fort it walks to without catching Pokemon
### Expected Behavior

Bot locates a fort, walks to it and collects loot along with any nearby pokemon.

### Actual Behavior

Bot begins walking to fort and then emits `Possibly searching too often, take a rest.` and then picks a new fort to walk to.

It then does this over and over again, see dump below:

```
2016-08-07 07:34:16,016 [  SpinFort] [INFO] [pokestop_searching_too_often] Possibly searching too often, take a rest.
2016-08-07 07:34:16,244 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Per Capita  - 0.12km (attraction of lure 0.12km)
2016-08-07 07:34:18,205 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Per Capita  - 0.10km (attraction of lure 0.11km)
2016-08-07 07:34:20,237 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Per Capita  - 0.09km (attraction of lure 0.11km)
2016-08-07 07:34:22,274 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Per Capita  - 0.08km (attraction of lure 0.10km)
2016-08-07 07:34:25,080 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Running Pinhead - 0.10km (attraction of lure 0.10km)
2016-08-07 07:34:27,285 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Running Pinhead - 0.08km (attraction of lure 0.08km)
2016-08-07 07:34:29,634 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Running Pinhead - 0.07km (attraction of lure 0.07km)
2016-08-07 07:34:32,235 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Running Pinhead - 0.06km (attraction of lure 0.06km)
2016-08-07 07:34:34,477 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Running Pinhead (3b716d7483d14ccf8eab6757d0b04a83.16)
2016-08-07 07:34:39,964 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Running Pinhead - 0.05km (attraction of lure 0.05km)
2016-08-07 07:34:42,114 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Running Pinhead (3b716d7483d14ccf8eab6757d0b04a83.16)
2016-08-07 07:34:47,673 [  SpinFort] [INFO] [pokestop_out_of_range] Pokestop Running Pinhead out of range.
2016-08-07 07:34:47,674 [  SpinFort] [INFO] [pokestop_searching_too_often] Possibly searching too often, take a rest.
2016-08-07 07:34:47,679 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Per Capita  - 0.08km (attraction of lure 0.33km)
2016-08-07 07:34:55,201 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Running Pinhead (3b716d7483d14ccf8eab6757d0b04a83.16)
2016-08-07 07:35:00,504 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Per Capita  - 0.07km (attraction of lure 0.32km)
2016-08-07 07:35:03,290 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Per Capita  - 0.05km (attraction of lure 0.32km)
2016-08-07 07:35:05,440 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Per Capita  - 0.04km (attraction of lure 0.32km)
2016-08-07 07:35:07,994 [  SpinFort] [INFO] [pokestop_out_of_range] Pokestop Per Capita  out of range.
2016-08-07 07:35:07,995 [  SpinFort] [INFO] [pokestop_searching_too_often] Possibly searching too often, take a rest.
2016-08-07 07:35:08,222 [MoveToFort] [INFO] [arrived_at_fort] Arrived at fort.
2016-08-07 07:35:08,223 [FollowSpiral] [INFO] [position_update] Walking from (-41.29176189044683, 174.78228744024358, 0) to (-41.29103333290947, 174.77899609590767, 0) (286.669511991 m)
2016-08-07 07:35:10,557 [  SpinFort] [INFO] [pokestop_out_of_range] Pokestop The Mover out of range.
2016-08-07 07:35:10,558 [  SpinFort] [INFO] [pokestop_searching_too_often] Possibly searching too often, take a rest.
2016-08-07 07:35:10,784 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Ban Shark Finning - 0.10km (attraction of lure 0.32km)
2016-08-07 07:35:13,088 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Ban Shark Finning - 0.09km (attraction of lure 0.31km)
2016-08-07 07:35:15,121 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Ban Shark Finning - 0.07km (attraction of lure 0.30km)
2016-08-07 07:35:17,148 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Ban Shark Finning - 0.06km (attraction of lure 0.29km)
2016-08-07 07:35:19,398 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Ban Shark Finning - 0.04km (attraction of lure 0.27km)
2016-08-07 07:35:21,741 [  SpinFort] [INFO] [pokestop_out_of_range] Pokestop Ban Shark Finning out of range.
2016-08-07 07:35:21,741 [  SpinFort] [INFO] [pokestop_searching_too_often] Possibly searching too often, take a rest.
2016-08-07 07:35:21,981 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Waitangi Park Graffiti Wall - 0.12km (attraction of lure 0.26km)
2016-08-07 07:35:24,262 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Waitangi Park Graffiti Wall - 0.11km (attraction of lure 0.24km)
2016-08-07 07:35:26,234 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Waitangi Park Graffiti Wall - 0.09km (attraction of lure 0.23km)
2016-08-07 07:35:28,479 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Waitangi Park Graffiti Wall - 0.08km (attraction of lure 0.21km)
2016-08-07 07:35:30,629 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Waitangi Park Graffiti Wall - 0.07km (attraction of lure 0.20km)
2016-08-07 07:35:32,927 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Waitangi Park Graffiti Wall - 0.05km (attraction of lure 0.19km)
2016-08-07 07:35:34,724 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Waitangi Park Graffiti Wall - 0.04km (attraction of lure 0.18km)
2016-08-07 07:35:37,398 [  SpinFort] [INFO] [pokestop_out_of_range] Pokestop Waitangi Park Graffiti Wall out of range.
2016-08-07 07:35:37,399 [  SpinFort] [INFO] [pokestop_searching_too_often] Possibly searching too often, take a rest.
2016-08-07 07:35:37,625 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Waitangi Gate  - 0.05km (attraction of lure 0.16km)
2016-08-07 07:35:40,254 [  SpinFort] [INFO] [pokestop_out_of_range] Pokestop Waitangi Gate  out of range.
2016-08-07 07:35:40,254 [  SpinFort] [INFO] [pokestop_searching_too_often] Possibly searching too often, take a rest.
2016-08-07 07:35:40,482 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Wellington Skate Park - 0.07km (attraction of lure 0.15km)
2016-08-07 07:35:42,826 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Wellington Skate Park - 0.05km (attraction of lure 0.14km)
2016-08-07 07:35:45,693 [  SpinFort] [INFO] [pokestop_out_of_range] Pokestop Wellington Skate Park out of range.
2016-08-07 07:35:45,694 [  SpinFort] [INFO] [pokestop_searching_too_often] Possibly searching too often, take a rest.
2016-08-07 07:35:45,917 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Spoked Pole - 0.06km (attraction of lure 0.12km)
2016-08-07 07:35:48,213 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Spoked Pole - 0.05km (attraction of lure 0.12km)
2016-08-07 07:35:50,434 [  SpinFort] [INFO] [pokestop_out_of_range] Pokestop Spoked Pole out of range.
2016-08-07 07:35:50,435 [  SpinFort] [INFO] [pokestop_searching_too_often] Possibly searching too often, take a rest.
2016-08-07 07:35:50,673 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Waitangi Playground - 0.11km (attraction of lure 0.11km)
2016-08-07 07:35:52,706 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Waitangi Playground - 0.09km (attraction of lure 0.09km)
2016-08-07 07:35:54,992 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Waitangi Playground - 0.08km (attraction of lure 0.08km)
2016-08-07 07:35:57,112 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Waitangi Playground - 0.07km (attraction of lure 0.07km)
2016-08-07 07:35:59,653 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Waitangi Playground - 0.05km (attraction of lure 0.05km)
2016-08-07 07:36:02,046 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Waitangi Playground (73f7bc6f0f564577823f611ef0c078bf.16)
2016-08-07 07:36:07,379 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Waitangi Playground - 0.04km (attraction of lure 0.04km)
2016-08-07 07:36:09,718 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Waitangi Playground (73f7bc6f0f564577823f611ef0c078bf.16)
2016-08-07 07:36:20,266 [  SpinFort] [INFO] [pokestop_out_of_range] Pokestop Waitangi Playground out of range.
2016-08-07 07:36:20,267 [  SpinFort] [INFO] [pokestop_searching_too_often] Possibly searching too often, take a rest.
2016-08-07 07:36:20,500 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Orbs of Light - 0.20km (attraction of lure 0.75km)
2016-08-07 07:36:22,887 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Waitangi Playground (73f7bc6f0f564577823f611ef0c078bf.16)
```

### Steps to Reproduce

Run bot as normal, after about 30 minutes this starts to happen for me.

### Other Information
OS: Debian 8
Git Commit: c81666d751b9ca8cd338760fdf920194fabf4ee4
Python Version: Python 2.7.9

### PR DESCRIPTION
**Pull request by geek-man**
Short Description: 

Fixes:
- #2698 
- 
- 

- Added api.activate_signature call to prevent issue after re-login.
- Also replaced deprecated log call with event_manager emit to prevent exception being thrown.

### CODE REVIEW

**Comment by geek-man** [comment]
@brantje FYI. I have created a new PR for the dev branch. Sorry about that!

File: pokemongo_bot/__init__.py [hunk 1]
```diff
@@ -547,11 +547,17 @@ def check_session(self, position):
...
+                    'api_error',
+                    sender=self,
+                    level='info',
+                    formatted='Session stale, re-logging in.'
+                )
                 position = self.position
                 self.api = ApiWrapper()
                 self.position = position
                 self.login()
+                self.api.activate_signature("encrypt.so")
```
Comment (brantje) [inline pokemongo_bot/__init__.py]: This is gonna give troube. Because it's gonna look in pokemongo_bot for encrypt.so.

**Comment by geek-man** [comment]
@brantje Re your note. I'd say you're right about that, but I'm just matching what the existing `_setup_api` function does. In my live copy I've just replaced all instances of "encrypt.so" with my full path.

Any preferred way of doing it? I would say it makes sense to have some code in there to find the absolute path of the file and refer to that path variable throughout the code. But just matching what's there for now.

I saw your PR #2768. Seems like a good way of doing it.

**Comment by brantje** [comment]
Still improving that pr. Maybe i'll make an function for it.

**Comment by Moonlight-Angel** [comment]
@geek-man Using @brantje 's method should be clean and bullet-proof now.

**Comment by geek-man** [comment]
@brantje Is it far to say that your PR for the OS detected loading method will get put in? If so, I'm about to go to bed, so I'm more than happy to edit my PR to use your function before I go, so this is compatible. 

I've just created a test branch and manually merged your PR into it. My initial testing seems fine, but I'm also going to wait for 30 minutes for the token to expire to 100% confirm this works when session re-logs in "for real".

**Comment by brantje** [comment]
@geek-man Yes, i think so. need some windows users to confirm that my PR is working on Windows 32bit / 64bit.

**Comment by thereyrey** [comment]
Im on windows 10. x64 and inserting this line:

Traceback (most recent call last):
  File "C:\Python27\PokemonGo-Bot\pokecli.py", line 41, in <module>
    from pokemongo_bot import PokemonGoBot, TreeConfigBuilder
  File "C:\Python27\PokemonGo-Bot\pokemongo_bot\__init__.py", line 555
    self.api.activate_signature("encrypt.so")
    ^
IndentationError: unexpected indent

Under _setup_api i use:
self.api.activate_signature("encrypt64bit.dll")
for the bot to run.
I've tried to use the same line where you inserted it, and gives the same error.

**Comment by geek-man** [comment]
@thereyrey Unexpected indent means that the spacing on that line of code is incorrect. Python requires that you make sure all lines are indented correctly, or it won't work.

Also make sure you're consistently using spaces. Using tabs in this case could cause problems.

**Comment by brantje** [comment]
@geek-man May i integrate your PR in my PR?

**Comment by thereyrey** [comment]
Went over the code again, removed some tabs, replacved with spaces. works.

**Comment by geek-man** [comment]
@brantje I've just added a commit to match your PR, but you can if you like.

**Comment by brantje** [comment]
Thanks, closing, my pr is merged, thanks for your contribution!

**Comment by geek-man** [comment]
@brantje I might be missing something, but it doesn't look like your merged changes has any of my changes? Dev branch doesn't have them either.

**Comment by edevil** [comment]
Yep, I don't see them on any branch.

**Comment by brantje** [comment]
Indeed, i guess they merged the branch before i could push.. o.. well you have the honor sir!

**Comment by krishven** [comment]
I changed the __init__.py file inline with the following commit. That's it right?

https://github.com/PokemonGoF/PokemonGo-Bot/commit/6960f356a8d841cefbfd1018b23877b73a6a0d12

**Pull request merged**